src/test/java/org/junitpioneer/jupiter/DefaultTimeZoneTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.TimeZone;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.test.event.ExecutionEventRecorder;
+import org.junitpioneer.AbstractPioneerTestEngineTests;
+
+@DisplayName("TimeZone extension")
+class DefaultTimeZoneTests extends AbstractPioneerTestEngineTests {
+
+	private static TimeZone TEST_DEFAULT_TIMEZONE;
+	private static TimeZone DEFAULT_TIMEZONE_BEFORE_TEST;
+
+	@BeforeAll
+	static void globalSetUp() {
+		DEFAULT_TIMEZONE_BEFORE_TEST = TimeZone.getDefault();
+		TimeZone utc = TimeZone.getTimeZone("UTC");
+		if (!DEFAULT_TIMEZONE_BEFORE_TEST.equals(utc)) {
+			TimeZone.setDefault(utc);
+		}
+		else {
+			TimeZone.setDefault(TimeZone.getTimeZone("GMT"));
+		}
+		TEST_DEFAULT_TIMEZONE = TimeZone.getDefault();
```

Comment:
I find this a little clearer (clarifies roles of UTC and GMT; no double-negative for else branch):

```java
// the extension sets UTC as test time zone unless it is already
// the system's time zone; in that case it uses GMT
DEFAULT_TIMEZONE_BEFORE_TEST = TimeZone.getDefault();
TimeZone utc = TimeZone.getTimeZone("UTC");
TimeZone gmt = TimeZone.getTimeZone("GMT");
if (DEFAULT_TIMEZONE_BEFORE_TEST.equals(utc)) {
	TimeZone.setDefault(gmt);
}
else {
	TimeZone.setDefault(utc);
}
TEST_DEFAULT_TIMEZONE = TimeZone.getDefault();
```